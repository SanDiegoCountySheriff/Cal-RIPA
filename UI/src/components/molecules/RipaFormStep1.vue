<template>
  <v-form ref="stepForm" lazy-validation>
    <ripa-officer v-model="model" toggle></ripa-officer>
    <ripa-stop-date v-model="model"></ripa-stop-date>
    <ripa-location
      v-model="model"
      :schools="schools"
      :beats="beats"
      :county-cities="countyCities"
      :last-location="lastLocation"
      :non-county-cities="nonCountyCities"
      :valid-last-location="validLastLocation"
    ></ripa-location>

    <v-spacer></v-spacer>

    <template v-if="!isFormValid">
      <v-alert type="error">
        Oops, you may have missed something! Please review your selections
        above.
      </v-alert>
    </template>

    <div class="tw-flex tw-mt-8 tw-justify-center">
      <v-btn outlined color="error" class="tw-mr-2" @click="handleCancel">
        Cancel
      </v-btn>
      <v-btn color="primary" @click="handleNext"> Next </v-btn>
    </div>
  </v-form>
</template>

<script>
import RipaOfficer from '@/components/molecules/RipaOfficer'
import RipaStopDate from '@/components/molecules/RipaStopDate'
import RipaLocation from '@/components/molecules/RipaLocation'
import RipaFormStepMixin from '@/components/mixins/RipaFormStepMixin'

export default {
  name: 'ripa-form-step1',

  mixins: [RipaFormStepMixin],

  components: { RipaOfficer, RipaStopDate, RipaLocation },

  props: {
    beats: {
      type: Array,
      default: () => [],
    },
    countyCities: {
      type: Array,
      default: () => [],
    },
    lastLocation: {
      type: Object,
      default: () => {},
    },
    nonCountyCities: {
      type: Array,
      default: () => {},
    },
    schools: {
      type: Array,
      default: () => [],
    },
    validLastLocation: {
      type: Boolean,
      default: false,
    },
  },
}
</script>
