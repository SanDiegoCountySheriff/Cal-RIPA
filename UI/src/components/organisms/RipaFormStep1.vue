<template>
  <v-form ref="stepForm" lazy-validation>
    <ripa-officer
      v-model="model.officer"
      @input="handleInput"
      toggle
    ></ripa-officer>
    <ripa-stop-date
      v-model="model.stopDetails"
      @input="handleInput"
    ></ripa-stop-date>
    <ripa-location
      v-model="model.location"
      :schools="schools"
      :beats="beats"
      :cities="cities"
      @input="handleInput"
    ></ripa-location>

    <v-spacer></v-spacer>

    <template v-if="!isValid">
      <v-alert type="error">
        Oops, you may have missed something! Please review your selections
        above.
      </v-alert>
    </template>

    <div class="tw-flex tw-mt-8 tw-justify-center">
      <v-btn outlined color="error" class="tw-mr-4" @click="handleCancel">
        Cancel
      </v-btn>
      <v-btn color="primary" class="tw-mr-4" @click="handleNext"> Next </v-btn>
    </div>
  </v-form>
</template>

<script>
import RipaOfficer from '@/components/molecules/RipaOfficer'
import RipaStopDate from '@/components/molecules/RipaStopDate'
import RipaLocation from '@/components/molecules/RipaLocation'
import RipaFormStepMixin from '@/components/mixins/RipaFormStepMixin'

export default {
  name: 'ripa-form-step1',

  mixins: [RipaFormStepMixin],

  components: { RipaOfficer, RipaStopDate, RipaLocation },

  data() {
    return {
      isValid: true,
      viewModel: {
        officer: this.value?.officer || null,
        stopDetails: this.value?.stopDetails || null,
        location: this.value?.location || null,
      },
    }
  },
}
</script>
