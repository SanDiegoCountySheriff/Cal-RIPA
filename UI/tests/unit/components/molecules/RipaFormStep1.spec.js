import RipaFormStep1 from '@/components/molecules/RipaFormStep1.vue'
import { shallowMount, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'

describe('Ripa Form Step 1', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  afterEach(() => {
    wrapper.destroy()
  })

  const factory = propsData => {
    return shallowMount(RipaFormStep1, {
      vuetify,
      propsData: {
        ...propsData,
        onOpenFavorites: jest.fn(),
        onOpenLastLocation: jest.fn(),
        onSaveFavorite: jest.fn(),
        onGpsLocation: jest.fn(),
        onUpdateUser: jest.fn(),
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = mount(RipaFormStep1, {
      vuetify,
      propsData: {
        onOpenFavorites: jest.fn(),
        onOpenLastLocation: jest.fn(),
        onSaveFavorite: jest.fn(),
        onGpsLocation: jest.fn(),
        onUpdateUser: jest.fn(),
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
