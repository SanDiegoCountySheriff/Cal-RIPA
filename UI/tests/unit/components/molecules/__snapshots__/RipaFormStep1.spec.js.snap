// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Form Step 1 should match snapshot 1`] = `
<v-form-stub lazyvalidation="true">
  <ripa-officer-stub onopenstatute="[Function]" user="[object Object]" onupdateuser="[Function]"></ripa-officer-stub>
  <ripa-user-dialog-stub user="[object Object]" onclose="[Function]" onsave="[Function]" adminediting="true" isrowkeydisabled="true" formtitle="Manage User"></ripa-user-dialog-stub>
  <!---->
  <ripa-stop-date-stub onopenstatute="[Function]" value="[object Object]"></ripa-stop-date-stub>
  <ripa-location-stub onopenstatute="[Function]" value="[object Object]" schools="" onopenfavorites="[Function]" onopenlastlocation="[Function]" onsavefavorite="[Function]" ongpslocation="[Function]"></ripa-location-stub>
  <v-spacer-stub tag="div"></v-spacer-stub>
  <!---->
  <div class="tw-flex tw-mt-8 tw-justify-center">
    <v-btn-stub color="error" outlined="true" tag="button" activeclass="" type="button" class="tw-mr-2">
      Cancel
    </v-btn-stub>
    <v-btn-stub color="primary" tag="button" activeclass="" type="button">
      Next
    </v-btn-stub>
  </div>
  <ripa-confirm-dialog-stub title="Confirm Continue" subtitle="This page may contain personally identifying information. Are you sure you want to continue?" onclose="[Function]" onconfirm="[Function]"></ripa-confirm-dialog-stub>
</v-form-stub>
`;
